app/actions/local/systems.ts

```diff
@@ -74,6 +83,148 @@ export async function storeConfig(serverUrl: string, config: ClientConfig | unde
+
+        // Do not run if clean up is already done today
+        if (lastRunAt && lastCleanedToday) {
+            return {error: undefined};
+        }
+
+        const globalPolicy = await getGlobalDataRetentionPolicy(database);
+        const granularPoliciesData = await getGranularDataRetentionPolicies(database);
+
+        // Get global data retention cutoff
+        let globalRetentionCutoff = 0;
+        if (globalPolicy?.message_deletion_enabled) {
+            globalRetentionCutoff = globalPolicy.message_retention_cutoff;
+        }
+
+        // Get Granular data retention policies
+        let teamPolicies: TeamDataRetentionPolicy[] = [];
+        let channelPolicies: ChannelDataRetentionPolicy[] = [];
+        if (granularPoliciesData) {
+            teamPolicies = granularPoliciesData.team;
+            channelPolicies = granularPoliciesData.channel;
+        }
+
+        const channelsCutoffs: {[key: string]: number} = {};
+
+        // Get channel level cutoff from team policies
+        for (let i = 0; i < teamPolicies.length; i++) {
+            const {team_id, post_duration} = teamPolicies[i];
+            // eslint-disable-next-line no-await-in-loop
+            const channelIds = await queryAllChannelsForTeam(database, team_id).fetchIds();
+            if (channelIds.length) {
+                const cutoff = getDataRetentionPolicyCutoff(post_duration);
+                channelIds.forEach((channelId) => {
+                    channelsCutoffs[channelId] = cutoff;
+                });
+            }
+        }
+
+        // Get channel level cutoff from channel policies
+        channelPolicies.forEach(({channel_id, post_duration}) => {
+            channelsCutoffs[channel_id] = getDataRetentionPolicyCutoff(post_duration);
+        });
+
+        const conditions = [];
+
+        const channelIds = Object.keys(channelsCutoffs);
+        if (channelIds.length) {
+            // Fetch posts by channel level cutoff
+            for (const channelId of channelIds) {
+                const cutoff = channelsCutoffs[channelId];
+                conditions.push(`(channel_id='${channelId}' AND create_at < ${cutoff})`);
+            }
+
+            // Fetch posts by global cutoff which are not already fetched by channel level cutoff
+            conditions.push(`(channel_id NOT IN ('${channelIds.join("','")}') AND create_at < ${globalRetentionCutoff})`);
+        } else {
+            conditions.push(`create_at < ${globalRetentionCutoff}`);
+        }
+
+        const postIds = await database.get<PostModel>(POST).query(
+            Q.unsafeSqlQuery(`SELECT * FROM ${POST} where ${conditions.join(' OR ')}`),
+        ).fetchIds();
+
+        if (postIds.length) {
+            const batchSize = 1000;
+            for (let i = 0; i < postIds.length; i += batchSize) {
+                const batch = postIds.slice(i, batchSize);
+                // eslint-disable-next-line no-await-in-loop
```

Comment:
Same as above

Reply:
I have changed in the other place but in here, I am not looping through the postIds instead looping through the `slices` of postIds.